Automattic/woocommerce-payments
# Cleanup the removed scripts rollout

Fixes https://github.com/Automattic/woocommerce-payments/issues/7909

#### Changes proposed in this Pull Request
After one of the recent cleanups, we [are using the old UPE split class](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments.php#L1320) to enqueue scripts. It extends `WC_Payments_Blocks_Payment_Method` which enqueues [everything needed for the checkout](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-blocks-payment-method.php#L56-L76). Right after, we're additionally trying to roll out [split UPE scipt](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-upe-split-blocks-payment-method.php#L20-L24) which 
* is not needed anymore
* was renamed to `blocks-checkout`

This led to `404` and because this script was not needed, there was no regression in terms of the functionality. This situation led me to the conclusion that we simply don't need `WC_Payments_UPE_Split_Blocks_Payment_Method` and we should use `WC_Payments_Blocks_Payment_Method` with its implementation of `get_payment_method_script_handles` that covers everything (and was also [executed previously](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-upe-split-blocks-payment-method.php#L18))










#### Testing instructions





* checkout `develop`, open Blocks checkout, confirm that you see `404` for upe split script in the network tab, while also `blocks-checkout.js` is rolled out ([screenshot](https://github.com/Automattic/woocommerce-payments/assets/22319444/c66c6a70-40d7-4b2e-86de-4a0d753c8bda))
* checkout `cleanup/redundant-script-enqueueing`, confirm no upe split is enqueued and `blocks-checkout.js` is still rolled out ([screenshot](https://github.com/Automattic/woocommerce-payments/assets/22319444/831d85a4-b381-4454-addc-3cb9a480e724))
* perform a card + non-card purchase with Blocks checkout as part of the sanity check


-------------------


**Post merge**
